clouddriver-kubernetes/src/main/groovy/com/netflix/spinnaker/clouddriver/kubernetes/v2/security/KubernetesV2Credentials.java

```diff
@@ -409,6 +443,28 @@ private boolean canReadKind(KubernetesKind kind, String checkNamespace) {
     }
   }
 
+  private boolean canReadKindRbac(KubernetesKind kind, String checkNamespace) throws Exception {
+    ApiClient client = Config.fromConfig(this.kubeconfigFile);
```

Comment:
Nice catch, I def didn't think about that. I'll see if there's any way I can incorporate that.

Reply:
@ezimanyi I dug your concerns about `kubectl` vs the API client. Based on what I've seen in the `kubectl` code [here](https://github.com/kubernetes/kubernetes/blob/master/pkg/kubectl/cmd/auth/cani.go) it doesn't look like it's doing anything different than what we were doing with `SelfSubjectAccessReview`. However, in my testing, the overhead of shelling out to `kubectl auth-can i` seems negligible vs using the API client directly.

In this chart, the yellow line is using the `kubectl` version instead of the API client:
![](https://cl.ly/6365249f46ad/download/[4b2ca51722c76d4db82fce71ac9bc19b]_Image%202019-07-15%20at%208.04.40%20AM.png)

We can still get a perf boost by using this new approach but continuing to shell out to `kubectl` for each call.